File: C++/missingNumber.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int sum = 0;
        for (int i = 0; i < nums.size(); ++i) {
            sum += nums[i];
        }

        int nRangeSum = 0;
        for (int i = 0; i < nums.size() + 1; ++i) {
            nRangeSum += i;
        }
        return nRangeSum - sum;
    }
};

File: Typescript/missingNumber.ts
function missingNumber(nums: number[]): number {
    let sum = 0;
    for (let i = 0; i < nums.length; ++i) {
        sum += nums[i];
    }

    let nRangeSum = 0;
    for (let i = 0; i < nums.length + 1; ++i) {
        nRangeSum += i;
    }
    return nRangeSum - sum;
}
